src/motion/features/viewport/types.ts

```diff
@@ -0,0 +1,24 @@
+import { RefObject } from "react"
+import { TargetAndTransition } from "../../../types"
+import { VariantLabels } from "../../types"
+
+export type ViewportEventHandler = () => void
+
+export interface ViewportOptions {
+    root?: RefObject<Element>
+    once?: boolean
+    margin?: string
+    amount?: "some" | "all" | number
+}
+
+export interface ViewportProps {
+    whileInView?: VariantLabels | TargetAndTransition
+    onViewportEnter?: ViewportEventHandler
+    onViewportLeave?: ViewportEventHandler
```

Comment:
While looking at our implementation, I noticed we provide a single callback that receives a boolean if the element is in viewport. I was wondering if that actually is a bit more flexible/smaller API than having two separate callbacks?

Reply:
`onViewportChange`?

Reply:
`onViewChange`

Reply:
I do like the idea of a single listener but I'm leaning towards keeping them enter/leave because it mirrors everything else, `onHoverStart`/`End`, `onPressStart`/`End` etc